docs/2-building-blocks-of-oop-part-1/hometask/src/Triangle.ts

```diff
@@ -0,0 +1,35 @@
+import { Shape } from "./Shape";
+import { Point } from "./Point";
+
+export class Triangle extends Shape {
+  constructor(pointA: Point, pointB: Point, pointC: Point);
+  constructor(pointA: Point, pointB: Point, pointC: Point, color: string, filled: boolean);
+  constructor(...args: any[]) {
+    const [pointA, pointB, pointC, color, filled] = args;
+    if (!pointA || !pointB || !pointC) {
+      throw new Error("Triangle should contain at least 3 points");
+    }
+    if (args.length === 1) {
+      super([pointA, pointB, pointC]);
+    }
```

Comment:
you can remove this `if` statement. The next call to `super` should handle missing `color` and `filled` parameters